src/System.Management.Automation/security/SecureStringHelper.cs

```diff
@@ -385,6 +385,21 @@ internal static SecureString Decrypt(string input, byte[] key, byte[] IV)
                 return s;
             }
         }
+
+#nullable enable
+        internal static unsafe SecureString FromPlainTextString(string plainTextString)
+        {
+            if (plainTextString.Length == 0)
+            {
+                return new SecureString();
+            }
```

Comment:
Nullable annotations do not still work in callsites and I'd add Assert to check null.

Reply:
> Nullable annotations do not still work in callsites and I'd add Assert to check null.

Apparently so, I was able to compile code with `SecureStringHelper.FromPlainTextString(null)`. In that case I would prefer to use `string.IsNullOrEmpty` to avoid possible NRE.

I wasn't aware the compiler didn't check nullability at callsites, is this by design? If so that certainly reduces the value of #12631.

Reply:
It checks if nullable annotations is enabled for callsites. So we get all benefits only after we finish annotations.
I agree with IsNullOrEmpty().